src/runtime_src/xdp/profile/device/aie_counter.h

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (C) 2019 Xilinx Inc - All rights reserved
+ * Xilinx Debug & Profile (XDP) APIs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef XDP_PROFILE_DEVICE_AIE_COUNTER_H
+#define XDP_PROFILE_DEVICE_AIE_COUNTER_H
+
+#include <stdexcept>
+#include "profile_ip_access.h"
+
+namespace xdp {
+
+/**
+ * AI Engine Counter ProfileIP (IP with safe access)
+ *
+ * Description:
+ *
+ * This class represents the high level exclusive and OS protected
+ * access to a single AIE counter on the device.
+ *
+ * Note:
+ *
+ * This class only aims at providing interface for easy and
+ * safe access to a single profiling IP. Managing the
+ * association between IPs and devices should be done in a
+ * different data structure that is built on top of this class.
+ */
+class AIECounter : public ProfileIP {
+public:
+    /**
+     * The constructor takes a device handle and a ip index
```

Comment:
```suggestion
     * The constructor takes a device handle and an ip index
```
also remove all the heading `*` in the comments